components/cli/pkg/commands/login.go

```diff
@@ -101,3 +125,158 @@ func RunLogin(registryURL string, username string, password string) {
 }
+
+// This function is to request credentials from the user
+func requestCredentials(username string) (string, string, error) {
+	initConf()
+	timeout := make(chan bool)
+	ch := make(chan string)
+	var code string
+	httpPortString := ":" + strconv.Itoa(loginConf.CallBackDefaultPort)
+	var codeReceiverPort = loginConf.CallBackDefaultPort
+	// This is to start the CLI auth in a different port is the default port is already occupied
+	for {
+		_, err := net.Dial("tcp", httpPortString)
+		if err != nil {
+			break
+		}
+		codeReceiverPort++
+		httpPortString = ":" + strconv.Itoa(codeReceiverPort)
+	}
+	var gitHubAuthUrl = "https://" + loginConf.IsHost + ":" + strconv.Itoa(loginConf.IsPort) +
+		"/oauth2/authorize?scope=openid&response_type=" + "code&redirect_uri=http%3A%2F%2F" +
+		loginConf.CallBackHost + "%3A" + strconv.Itoa(codeReceiverPort) +
+		"%2Fauth&client_id=" + loginConf.SpClientId + "&fidp=google"
+	fmt.Println("\n", gitHubAuthUrl)
+	go func() {
+		mux := http.NewServeMux()
+		server := http.Server{Addr: httpPortString, Handler: mux}
+		//var timer *time.Timer
+		mux.HandleFunc(loginConf.CallBackContextPath, func(w http.ResponseWriter, r *http.Request) {
+			err := r.ParseForm()
+			if err != nil {
+				util.ExitWithErrorMessage("Error parsing the code", err)
+			}
+			code = r.Form.Get(loginConf.CallBackParameter)
+			ch <- code
+			if err != nil {
+				fmt.Printf("Error occurred while establishing the server: %s\n", err)
+				return
+			}
+			if len(code) != 0 {
+				http.Redirect(w, r, "https://"+loginConf.IsHost+":"+strconv.Itoa(loginConf.IsPort)+
+					"/authenticationendpoint/auth_success.html", http.StatusSeeOther)
+			} else {
+				http.Redirect(w, r, "fail", http.StatusSeeOther)
+			}
+			flusher, ok := w.(http.Flusher)
+			if ok {
+				flusher.Flush()
+			}
+			err = server.Shutdown(context.Background())
+			if err != nil {
+				util.ExitWithErrorMessage("Error while shutting down the server\n", err)
+			}
+		})
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			util.ExitWithErrorMessage("Error while establishing the service", err)
+		}
+	}()
+
+	err := util.OpenBrowser(gitHubAuthUrl)
+	if err != nil {
+		fmt.Printf("Could not resolve the given url %s. Started to operate in the healess "+
+			"mode\n", gitHubAuthUrl)
+		return handleHeadless(username)
+	}
+	// Setting up a timeout
+	go func() {
+		time.Sleep(5 * time.Minute)
+		timeout <- true
+	}()
+	// Wait for a code, or timeout
+	select {
+	case <-ch:
+	case <-timeout:
+		fmt.Println("Time out. Did not receive any code")
+		os.Exit(0)
+	}
+	token := getTokenFromCode(code, codeReceiverPort)
+	username, idToken := getSubjectAndUsernameFromJWT(token)
+	return username, idToken, nil
+}
+
+// This function is to allow this login flow to work in headless mode
+func handleHeadless(username string) (string, string, error) {
+	var password string
+	if username == "" {
+		var err error
```

Comment:
We don't need to have a separate err definition. Please use `:=` when assigning itself. It's the normal practice in Go.